### PR TITLE
Do not relocate vector table on ARMv8-R aarch32

### PR DESCRIPTION
**Pull request by julien-massot**
ARMv8-R allows to set the vector table address using VBAR
register, so there is no need to relocate it.

Move away vector_table setting from reset.S and move it to
relocate vector table function as it's done for Cortex-M
CPU.

Signed-off-by: Julien Massot <julien.massot@iot.bzh>

### CODE REVIEW

File: include/arch/arm/aarch32/cortex_a_r/cpu.h
Comment (stephanosio) [inline include/arch/arm/aarch32/cortex_a_r/cpu.h:51]:
```suggestion
#define VBAR_Msk (0xFFFFFFE0U)
```

It would also be nice to keep it consistent with the rest in this file.
```suggestion
#define VBAR_MASK		(0xFFFFFFE0U)
```

Comment (julien-massot) [reply include/arch/arm/aarch32/cortex_a_r/cpu.h]: Thanks @stephanosio looks like I misread documentation. Bits [4:0] Reserved, RES0. bit(4) included :)

File: include/arch/arm/aarch32/cortex_a_r/cpu.h
Comment (stephanosio) [inline include/arch/arm/aarch32/cortex_a_r/cpu.h:51]: It would be nice if this is named `VBAR_MASK` like the rest in this file (e.g. `DFSR_WRITE_MASK`, `MPIDR_AFFLVL_MASK`, ...), but I leave this up to the maintainer.

**Review by stephanosio — DISMISSED** [review]

File: include/arch/arm/aarch32/cortex_a_r/cpu.h
Comment (julien-massot) [reply include/arch/arm/aarch32/cortex_a_r/cpu.h:51]: You're right I tried to make it consistent with the CONFIG_CPU_CORTEX_M_HAS_VTOR case, but VBAR_MASK is better

**Review by stephanosio — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Comment by carlescufi** [comment]
@povergoing could you please take a look?

**Review by bbolen — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

**Pull request merged**
